src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/RazorHtmlCSharpLanguageServer.cs

```diff
@@ -191,6 +216,17 @@ public Task<Location[]> GoToImplementationAsync(TextDocumentPositionParams posit
             return ExecuteRequestAsync<TextDocumentPositionParams, Location[]>(Methods.TextDocumentImplementationName, positionParams, _clientCapabilities, cancellationToken);
         }
 
+        [JsonRpcMethod("textDocument/semanticTokens", UseSingleObjectParameterDeserialization = true)]
```

Comment:
Why can't our endpoint on our language server suffice? Not sure why we need to proxy the request. I wont leave any more comments about this but I have the same concern with the `SemanticTokenHandler`

Reply:
I'm not exactly sure why, but that seemed to be the pattern established with all the  other endpoints. It seems like you  created this class in the first place then the rest of us added on over time. Do you have additional context on this or know who might have more?

Reply:
Talked offline, this is needed when results may be needed from both Razor and C#, Semantic doesn't do that yet so we don't need it yet.